source/common/grpc/async_client_impl.h

```diff
@@ -185,9 +215,10 @@ class AsyncClientStreamImpl : public AsyncClientStream<RequestType>,
 
     // This will destroy us, but only do so if we are actually in a list. This does not happen in
     // the immediate failure case.
-    if (LinkedObject<AsyncClientStreamImpl<RequestType, ResponseType>>::inserted()) {
-      LinkedObject<AsyncClientStreamImpl<RequestType, ResponseType>>::removeFromList(
-          parent_.active_streams_);
+    if (LinkedObject<AsyncStreamImpl<RequestType, ResponseType>>::inserted()) {
+      parent_.dispatcher_.deferredDelete(
```

Comment:
You probably already looked at this, but just double check that the stream/request does not reference client in its destructor, or there might be ordering issues depending on how client is destructed. (I haven't looked at code but just pointing this out from past experience).